src/execution/operator/persistent/physical_insert.cpp

```diff
@@ -256,19 +252,49 @@ idx_t PhysicalInsert::PerformOnConflictAction(ExecutionContext &context, DataChu
 	ExpressionExecutor executor(context.client, set_expressions);
 	executor.Execute(chunk, update_chunk);
 	update_chunk.SetCardinality(chunk);
+}
+
+idx_t PhysicalInsert::PerformGlobalOnConflictAction(ExecutionContext &context, DataChunk &chunk,
+                                                    TableCatalogEntry &table, Vector &row_ids) const {
+	if (action_type == OnConflictAction::NOTHING) {
+		return 0;
+	}
+
+	DataChunk update_chunk;
+	CreateUpdateChunk(context, chunk, table, row_ids, update_chunk);
 
 	auto &data_table = table.GetStorage();
 	// Perform the update, using the results of the SET expressions
 	data_table.Update(table, context.client, row_ids, set_columns, update_chunk);
 	return update_chunk.size();
 }
 
+idx_t PhysicalInsert::PerformLocalOnConflictAction(ExecutionContext &context, DataChunk &chunk,
+                                                   TableCatalogEntry &table, Vector &row_ids) const {
+	if (action_type == OnConflictAction::NOTHING) {
+		return 0;
+	}
+
+	DataChunk update_chunk;
+	CreateUpdateChunk(context, chunk, table, row_ids, update_chunk);
+
+	auto &data_table = table.GetStorage();
+	auto &local_storage = LocalStorage::Get(context.client, data_table.db);
+	// Perform the update, using the results of the SET expressions
+	local_storage.Update(data_table, row_ids, set_columns, update_chunk);
+	return update_chunk.size();
+}
+
 // TODO: should we use a hash table to keep track of this instead?
-void PhysicalInsert::RegisterUpdatedRows(InsertLocalState &lstate, const Vector &row_ids, idx_t count) const {
+void PhysicalInsert::RegisterUpdatedRows(InsertLocalState &lstate, const Vector &row_ids, idx_t count,
+                                         bool global) const {
 	// Insert all rows, if any of the rows has already been updated before, we throw an error
 	auto data = FlatVector::GetData<row_t>(row_ids);
+
+	// Are these row ids refering to the transaction-local storage or the global storage?
```

Comment:
Small nit: this comment reads a bit odd; maybe rephrase it as a statement instead of a question?